base/promotion.jl

```diff
@@ -222,9 +222,8 @@ minmax(x::Real, y::Real) = minmax(promote(x, y)...)
 # for the multiplication of two types,
 #   promote_op{R<:MyType,S<:MyType}(::typeof(*), ::Type{R}, ::Type{S}) = MyType{multype(R,S)}
 promote_op(::Any)    = (@_pure_meta; Bottom)
-promote_op(::Any, T) = (@_pure_meta; T)
-promote_op{T}(::Type{T}, ::Any) = (@_pure_meta; T)
```

Comment:
I've added it back, that's an important special case and it doesn't create any issues.